dts/arm/nxp/nxp_rt.dtsi

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) 2017, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arm/armv7-m.dtsi>
+#include <dt-bindings/clock/imx_ccm.h>
+#include <dt-bindings/i2c/i2c.h>
+
+/ {
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu@0 {
+			device_type = "cpu";
+			compatible = "arm,cortex-m7";
+			reg = <0>;
+		};
+	};
+
+	soc {
+		flexram0: flexram@400b0000 {
+			compatible = "nxp,imx-flexram";
+			reg = <0x400b0000 0x4000>;
+			interrupts = <38 0>;
+
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			itcm0: itcm0 {
+				reg = <0x00000000 0x20000>;
+			};
+
+			dtcm0: dtcm0 {
+				reg = <0x20000000 0x20000>;
+			};
+
+			ocram0: ocram0 {
+				reg = <0x20200000 0x40000>;
+			};
+		};
+
+		ccm: ccm@400fc000 {
+			compatible = "nxp,imx-ccm";
+			reg = <0x400fc000 0x4000>;
+			label = "CCM";
+
+			clock-controller;
+			#clocks-cells = <3>;
+		};
+
+		gpio1: gpio@401b8000 {
+			compatible = "nxp,imx-gpio";
+			reg = <0x401b8000 0x4000>;
+			interrupts = <80 0>, <81 0>;
+			label = "GPIO_1";
+		};
+
+		gpio2: gpio@401bc000 {
+			compatible = "nxp,imx-gpio";
+			reg = <0x401bc000 0x4000>;
+			interrupts = <82 0>, <83 0>;
+			label = "GPIO_2";
+		};
+
+		gpio3: gpio@401c0000 {
+			compatible = "nxp,imx-gpio";
+			reg = <0x401c0000 0x4000>;
+			interrupts = <84 0>, <85 0>;
+			label = "GPIO_3";
+		};
+
+		gpio4: gpio@401c4000 {
+			compatible = "nxp,imx-gpio";
+			reg = <0x401c4000 0x4000>;
+			interrupts = <86 0>, <87 0>;
+			label = "GPIO_4";
+		};
+
+		gpio5: gpio@400c0000 {
+			compatible = "nxp,imx-gpio";
+			reg = <0x400c0000 0x4000>;
+			interrupts = <88 0>, <89 0>;
+			label = "GPIO_5";
+		};
+
+		i2c1: i2c@403f0000 {
+			compatible = "nxp,kinetis-lpi2c";
```

Comment:
do we have a yaml & driver for this?  If not lets remove the node til we do.

Reply:
I have one in the works and originally intended to include it in this PR, but it's not working correctly so I pulled it out.